app/src/getters/getTransactionalClient.ts

```diff
@@ -0,0 +1,39 @@
+import { type StreamrClient, type ExternalProvider } from 'streamr-client'
+import Web3 from 'web3'
+import getWeb3 from '../utils/web3/getWeb3'
+import getChainId from '../utils/web3/getChainId'
+
+type ArgType<T> = T extends (arg: infer R) => any ? R : never
+
+let chainId: number | undefined
+
+type Provider = ArgType<typeof Web3.prototype.setProvider> & { chainId?: typeof chainId }
+
+let streamrClient: StreamrClient | undefined
+
+let provider: Provider | undefined
+
+export default async function getTransactionalClient() {
+    const { currentProvider } = getWeb3()
+
+    const currentChainId = await getChainId()
+
+    if (streamrClient && currentProvider === provider && currentChainId === chainId) {
+        return streamrClient
+    }
+
+    chainId = currentChainId
+
+    provider = currentProvider
+
+    streamrClient = new (await require('streamr-client')).StreamrClient({
+        auth: {
+            ethereum: currentProvider as ExternalProvider,
+        },
+    })
```

Comment:
@pmichalowski @tumppi apparently our webpack config (guessing here) doesn't like the client imported directly and complains in tests, out of all places.

This is better anyway, the async way, because `streamr-client-react` doesn't import the library directly either (dynamic import there).

🤞🏻 it builds nicely.